### PR TITLE
Replace _checkPanoptesSession with _getNewToken

### PR DESCRIPTION
**Pull request by eatyourgreens**
Replaces `_checkPanoptesSession()` with `_getNewToken()`, which attempts to get a new token from Panoptes via an iframe (and should do only that.) Stored tokens are not returned, since we should only be calling this method when the stored token is about to expire.

Fixes a problem mentioned in https://github.com/zooniverse/shakespeares_world/issues/366#issuecomment-362587573 which is that the client would always return an expired token, if it was stored in session storage.

Also fixes a bug where `checkBearerToken` throws an error when getting a new token, because `this._deleteBearerToken` is undefined in the catch block here https://github.com/zooniverse/panoptes-javascript-client/blob/595e694c71dd877ddbae4a13e670fc466314258e/lib/oauth.js#L279-L284

### CODE REVIEW

**Comment by simoneduca** [comment]
Is that the same problem I was refer to in your older PR https://github.com/zooniverse/panoptes-javascript-client/pull/85#issuecomment-368518881?

**Comment by eatyourgreens** [comment]
Could be. It's hard to tell, since the token issued by Panoptes is valid for 2 hours and the console logs don't tell us if that token came from a network call to Panoptes, or from session storage.

**Comment by eatyourgreens** [comment]
No wait, 'Found existing Panoptes session' is printed out when a token is obtained from the iframe URL. https://github.com/zooniverse/panoptes-javascript-client/blob/595e694c71dd877ddbae4a13e670fc466314258e/lib/oauth.js#L201-L204

**Comment by shaunanoordin** [comment]
The code update looks good and my initial testing looks OK so far, but I was trying to follow up by checking out if there were any other function()s that require `.this` that weren't bind()-ed properly.

And then I saw how the .bind()s were use in the lib components and thought, 😳 OK, never mind then. I'll continue with standard user testing, but out of curiosity, why does the PJC bind functions when they're being called/returned instead of binding at the function declaration, or equivalent?

**Comment by eatyourgreens** [comment]
> Sometimes when I'm writing Javascript I want to throw up my hands and say "this is bullshit!" but I can never remember what "this" refers to. 
https://twitter.com/bendhalpern/status/578925947245633536?s=21

The use of .bind(), particularly nested bound calls, makes some of the auth code really hard to follow. 

Without it, `this` would refer to the calling promise in those anonymous callbacks. Your comment does make me think the refresh code would be easier to follow as
```javascript
_getNewToken()
  .then(this._handleNewBearerToken)
  .catch(this._deleteBearerToken);
```

which would get rid of a couple of those .bind() calls.

**Comment by simoneduca** [comment]
I'll wait until tomorrow before merging to see if @camallen wants to add anything to this.

**Comment by eatyourgreens** [comment]
I've added a couple of changes that should keep your Auth headers intact until the Panoptes session has actually expired, so you can continue to make API calls in the last 5 minutes of your session.

**Comment by eatyourgreens** [comment]
No problems here either, browsing SW on localhost in Firefox with no restrictions and Chrome with third-party cookies blocked.
